src/Doxyfile.in

```diff
@@ -978,6 +1040,25 @@ USE_HTAGS              = NO
 
 VERBATIM_HEADERS       = YES
 
+# If the CLANG_ASSISTED_PARSING tag is set to YES then doxygen will use the
+# clang parser (see: http://clang.llvm.org/) for more accurate parsing at the
+# cost of reduced performance. This can be particularly helpful with template
+# rich C++ code for which doxygen's built-in parser lacks the necessary type
+# information.
+# Note: The availability of this option depends on whether or not doxygen was
+# generated with the -Duse-libclang=ON option for CMake.
+# The default value is: NO.
+
+CLANG_ASSISTED_PARSING = NO
```

Comment:
I wonder if that would have any effect on our code, or if the functions we document are simple enough that it doesn't matter.

Reply:
I tried with clang with `CLANG_ASSISTED_PARSING = YES` and `CLANG_OPTIONS = -I/my/path/to/CGAL-5.2/include  -I/usr/include/eigen3/` and I faced this error:
```bash
Searching for files in directory /my/path/to/gudhi-deverror: /home/vincent/workspace/gudhi-devel/build/CMakeFiles/3.10.2/CompilerIdC/CMakeCCompilerId.c:2:3: error: "A C++ compiler has been selected for C." [clang]
```
The final result does not seem to have additional information. By looking closer at a diff, the `*.js` seems to provide more links.
But I also notified that the sources in `Files` appeared empty at visualization, whereas the html seems to be well filled when this option is set. To be investigated.

The most notable difference is the time to generate the documentation with this option, that reaches about 13 minutes on my computer, against few seconds without this option. But maybe because of my configuration (cf. error above).